client/gateway.go

```diff
@@ -770,12 +875,181 @@ func (cli *VanClient) GatewayInit(ctx context.Context, gatewayName string, gatew
+			for _, deleted := range bcDiff.TcpConnectors.Deleted {
+				if err = agent.Delete(getEntity("tcp", gatewayEgress), deleted); err != nil {
+					return gatewayName, fmt.Errorf("Error removing entity connector: %w", err)
+				}
+			}
+			for _, added := range bcDiff.TcpConnectors.Added {
+				record := map[string]interface{}{}
+				added.SiteId = routerId
+				if err = convert(added, &record); err != nil {
+					return gatewayName, fmt.Errorf("Failed to convert record: %w", err)
+				}
+				if err = agent.Create(getEntity("tcp", gatewayEgress), added.Name, record); err != nil {
+					return gatewayName, fmt.Errorf("Error adding tcp entity: %w", err)
+				}
+			}
+
+			for _, deleted := range bcDiff.TcpListeners.Deleted {
+				if err = agent.Delete(getEntity("tcp", gatewayIngress), deleted); err != nil {
+					return gatewayName, fmt.Errorf("Error removing entity listener: %w", err)
+				}
+			}
+			for _, added := range bcDiff.TcpListeners.Added {
+				record := map[string]interface{}{}
+				added.SiteId = routerId
+				if err = convert(added, &record); err != nil {
+					return gatewayName, fmt.Errorf("Failed to convert record: %w", err)
+				}
+				if err = agent.Create(getEntity("tcp", gatewayIngress), added.Name, record); err != nil {
+					return gatewayName, fmt.Errorf("Error adding tcp entity: %w", err)
+				}
+			}
+
+			for _, deleted := range bcDiff.HttpConnectors.Deleted {
+				if err = agent.Delete(getEntity("http", gatewayEgress), deleted); err != nil {
+					return gatewayName, fmt.Errorf("Error removing entity connector: %w", err)
+				}
+			}
+			for _, added := range bcDiff.HttpConnectors.Added {
+				record := map[string]interface{}{}
+				added.SiteId = routerId
+				if err = convert(added, &record); err != nil {
+					return gatewayName, fmt.Errorf("Failed to convert record: %w", err)
+				}
+				if err = agent.Create(getEntity("http", gatewayEgress), added.Name, record); err != nil {
+					return gatewayName, fmt.Errorf("Error adding tcp entity: %w", err)
+				}
+			}
+
+			for _, deleted := range bcDiff.HttpListeners.Deleted {
+				if err = agent.Delete(getEntity("http", gatewayIngress), deleted); err != nil {
+					return gatewayName, fmt.Errorf("Error removing entity listener: %w", err)
+				}
+			}
+			for _, added := range bcDiff.HttpListeners.Added {
+				record := map[string]interface{}{}
+				added.SiteId = routerId
+				if err = convert(added, &record); err != nil {
+					return gatewayName, fmt.Errorf("Failed to convert record: %w", err)
+				}
+				if err = agent.Create(getEntity("http", gatewayIngress), added.Name, record); err != nil {
+					return gatewayName, fmt.Errorf("Error adding http entity: %w", err)
+				}
+			}
+
+			routerConfigCurrent.Bridges = *bridgeConfigFromFile
+			routerConfigCurrent.UpdateConfigMap(configmap)
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				_, err = cli.KubeClient.CoreV1().ConfigMaps(cli.GetNamespace()).Update(configmap)
+				return err
+			})
```

Comment:
@ajssmith don't you need to include these two lines 1023:1024 inside the `RetryOnConflict` block,
along with a "reload" of the original `configmap` variable (to make sure you're on the latest version)?

Reply:
@fgiorgetti good observation, all retry logic needs re-factoring and will update